test/Dialect/FIRRTL/extract-instances.mlir

```diff
@@ -0,0 +1,129 @@
+// RUN: circt-opt --firrtl-extract-instances --verify-diagnostics %s | FileCheck %s
+
+// Tests extracted from:
+// - test/scala/firrtl/ExtractBlackBoxes.scala
+
+//===----------------------------------------------------------------------===//
+// ExtractBlackBoxes Simple
+//===----------------------------------------------------------------------===//
+
+// CHECK: firrtl.circuit "ExtractBlackBoxesSimple"
+firrtl.circuit "ExtractBlackBoxesSimple" attributes {annotations = [{class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "BlackBoxes"}]} {
```

Comment:
The test could be simplified by removing the clock/reset ports.

Reply:
I've created a separate test case that does that, but kept this to match the original Scala FIRRTL compiler test. But good point, I'll do an overall cleanup pass at the end to throw out the unused clock/reset stuff :+1: